allennlp/common/testing/model_test_case.py

```diff
@@ -133,16 +133,25 @@ def check_model_computes_gradients_correctly(model, model_batch):
         model.zero_grad()
         result = model(**model_batch)
         result["loss"].backward()
-
-        for parameter in model.parameters():
+        has_zero_or_none_grads = {}
+        for name, parameter in model.named_parameters():
             zeros = torch.zeros(parameter.size())
             if parameter.requires_grad:
+
+                if parameter.grad is None:
+                    has_zero_or_none_grads[name] = "No gradient computed (i.e parameter.grad is None)"
                 # Some parameters will only be partially updated,
                 # like embeddings, so we just check that any gradient is non-zero.
-                assert (parameter.grad.data.cpu() != zeros).any()
+                if not (parameter.grad.data.cpu() != zeros).any():
```

Comment:
is there a way to rewrite this? `not (grad != zeros).any()` is really hard to understand

Reply:
like, would

```
if (parameter.grad.data.cpu() == 0).all()
```

work?